### ISSUE TITLE
everytime i import nikto scan i get Server Error (500) same for trufflehog

### ISSUE THREAD

**Issue by AliElKhatteb** [DefectDojo/django-DefectDojo#3268]
**Bug description**
A clear and concise description of what the bug is. For errors include at least the exact error message you are seeing (including traceback).

**Steps to reproduce**
1. Go to 'finding'
2. Click on 'import scan results'
3. scan type nikto scan
4. import

**Expected behavior**
its should be import but its not
**Deployment method** *(select with an `X`)*
- [ ] Kubernetes
- [ X] Docker
- [ ] setup.bash / legacy-setup.bash

**Environment information**
 - Operating System: [e.g. Ubuntu 20.04]

**Sample scan files** (optional)
?xml version="1.0" ?>
<!DOCTYPE niktoscan SYSTEM "/usr/share/doc/nikto/nikto.dtd">
<niktoscan hoststest="0" options="-h 127.0.0.1 -p 8070 -output nikto-output.xml" version="2.1.5" scanstart="Mon Nov 16 03:45:06 2020" scanend="Wed Dec 31 19:00:00 1969" scanelapsed=" seconds" nxmlversion="1.2">

<scandetails targetip="127.0.0.1" targethostname="localhost" targetport="8070" targetbanner="" starttime="2020-11-16 03:45:07" sitename="http://localhost:8070" siteip="http://127.0.0.1:8070" hostheader="localhost" errors="0" checks="6544">

<item id="999976" osvdbid="0" osvdblink="http://osvdb.org/0" method="GET">
<description><![CDATA[The anti-clickjacking X-Frame-Options header is not present.]]></description>
<uri><![CDATA[/]]></uri>
<namelink><![CDATA[http://localhost:8070/]]></namelink>
<iplink><![CDATA[http://127.0.0.1:8070/]]></iplink>
</item>

<item id="999984" osvdbid="0" osvdblink="http://osvdb.org/0" method="GET">
<description><![CDATA[Server leaks inodes via ETags, header found with file /favicon.ico, fields: 0xW/21630 0x1602011686000 ]]></description>
<uri><![CDATA[/favicon.ico]]></uri>
<namelink><![CDATA[http://localhost:8070/favicon.ico]]></namelink>
<iplink><![CDATA[http://127.0.0.1:8070/favicon.ico]]></iplink>
</item>

<item id="500008" osvdbid="39272" osvdblink="http://osvdb.org/39272" method="GET">
<description><![CDATA[favicon.ico file identifies this server as: Apache Tomcat]]></description>
<uri><![CDATA[/favicon.ico]]></uri>
<namelink><![CDATA[]]></namelink>
<iplink><![CDATA[http://:/favicon.ico]]></iplink>
</item>

<item id="999990" osvdbid="0" osvdblink="http://osvdb.org/0" method="OPTIONS">
<description><![CDATA[Allowed HTTP Methods: GET, HEAD, POST, PUT, DELETE, OPTIONS ]]></description>
<uri><![CDATA[/]]></uri>
<namelink><![CDATA[http://localhost:8070/]]></namelink>
<iplink><![CDATA[http://127.0.0.1:8070/]]></iplink>
</item>

<item id="999978" osvdbid="397" osvdblink="http://osvdb.org/397" method="GET">
<description><![CDATA[HTTP method ('Allow' Header): 'PUT' method could allow clients to save files on the web server.]]></description>
<uri><![CDATA[/]]></uri>
<namelink><![CDATA[http://localhost:8070/]]></namelink>
<iplink><![CDATA[http://127.0.0.1:8070/]]></iplink>
</item>

<item id="999976" osvdbid="5646" osvdblink="http://osvdb.org/5646" method="GET">
<description><![CDATA[HTTP method ('Allow' Header): 'DELETE' may allow clients to remove files on the web server.]]></description>
<uri><![CDATA[/]]></uri>
<namelink><![CDATA[http://localhost:8070/]]></namelink>
<iplink><![CDATA[http://127.0.0.1:8070/]]></iplink>
</item>

<item id="000366" osvdbid="0" osvdblink="http://osvdb.org/0" method="GET">
<description><![CDATA[/examples/servlets/index.html: Apache Tomcat default JSP pages present.]]></description>
<uri><![CDATA[/examples/servlets/index.html]]></uri>
<namelink><![CDATA[http://localhost:8070/examples/servlets/index.html]]></namelink>
<iplink><![CDATA[http://127.0.0.1:8070/examples/servlets/index.html]]></iplink>
</item>

<item id="000834" osvdbid="3931" osvdblink="http://osvdb.org/3931" method="GET">
<description><![CDATA[/myphpnuke/links.php?op=MostPopular&ratenum=[script]alert(document.cookie);[/script]&ratetype=percent: myphpnuke is vulnerable to Cross Site Scripting (XSS). CA-2000-02.]]></description>
<uri><![CDATA[/myphpnuke/links.php?op=MostPopular&ratenum=[script]alert(document.cookie);[/script]&ratetype=percent]]></uri>
<namelink><![CDATA[http://localhost:8070/myphpnuke/links.php?op=MostPopular&ratenum=[script]alert(document.cookie);[/script]&ratetype=percent]]></namelink>
<iplink><![CDATA[http://127.0.0.1:8070/myphpnuke/links.php?op=MostPopular&ratenum=[script]alert(document.cookie);[/script]&ratetype=percent]]></iplink>
</item>

<item id="000863" osvdbid="4598" osvdblink="http://osvdb.org/4598" method="GET">
<description><![CDATA[/members.asp?SF=%22;}alert(223344);function%20x(){v%20=%22: Web Wiz Forums ver. 7.01 and below is vulnerable to Cross Site Scripting (XSS). CA-2000-02.]]></description>
<uri><![CDATA[/members.asp?SF=%22;}alert(223344);function%20x(){v%20=%22]]></uri>
<namelink><![CDATA[http://localhost:8070/members.asp?SF=%22;}alert(223344);function%20x(){v%20=%22]]></namelink>
<iplink><![CDATA[http://127.0.0.1:8070/members.asp?SF=%22;}alert(223344);function%20x(){v%20=%22]]></iplink>
</item>

<item id="000888" osvdbid="2946" osvdblink="http://osvdb.org/2946" method="GET">
<description><![CDATA[/forum_members.asp?find=%22;}alert(9823);function%20x(){v%20=%22: Web Wiz Forums ver. 7.01 and below is vulnerable to Cross Site Scripting (XSS). CA-2000-02.]]></description>
<uri><![CDATA[/forum_members.asp?find=%22;}alert(9823);function%20x(){v%20=%22]]></uri>
<namelink><![CDATA[http://localhost:8070/forum_members.asp?find=%22;}alert(9823);function%20x(){v%20=%22]]></namelink>
<iplink><![CDATA[http://127.0.0.1:8070/forum_members.asp?find=%22;}alert(9823);function%20x(){v%20=%22]]></iplink>
</item>

<item id="999960" osvdbid="0" osvdblink="http://osvdb.org/0" method="GET">
<description><![CDATA[Cookie JSESSIONID created without the httponly flag]]></description>
<uri><![CDATA[/examples/jsp/snp/snoop.jsp]]></uri>
<namelink><![CDATA[http://localhost:8070/examples/jsp/snp/snoop.jsp]]></namelink>
<iplink><![CDATA[http://127.0.0.1:8070/examples/jsp/snp/snoop.jsp]]></iplink>
</item>

<item id="001355" osvdbid="3720" osvdblink="http://osvdb.org/3720" method="GET">
<description><![CDATA[/examples/jsp/snp/snoop.jsp: Displays information about page retrievals, including other users.]]></description>
<uri><![CDATA[/examples/jsp/snp/snoop.jsp]]></uri>
<namelink><![CDATA[http://localhost:8070/examples/jsp/snp/snoop.jsp]]></namelink>
<iplink><![CDATA[http://127.0.0.1:8070/examples/jsp/snp/snoop.jsp]]></iplink>
</item>

<item id="006525" osvdbid="0" osvdblink="http://osvdb.org/0" method="GET">
<description><![CDATA[/manager/html: Default Tomcat Manager interface found]]></description>
<uri><![CDATA[/manager/html]]></uri>
<namelink><![CDATA[http://localhost:8070/manager/html]]></namelink>
<iplink><![CDATA[http://127.0.0.1:8070/manager/html]]></iplink>
</item>

<statistics elapsed="37" itemsfound="12" itemstested="6544" endtime="2020-11-16 03:45:44" />
</scandetails>

</niktoscan>

**Screenshots** (optional)

**Console logs** (optional)
'NoneType' object has no attribute 'group'
Traceback (most recent call last):
  File "./dojo/engagement/views.py", line 624, in import_scan_results
    parser = import_parser_factory(file, t, active, verified)
  File "./dojo/tools/factory.py", line 117, in import_parser_factory
    parser = NiktoXMLParser(file, test)
  File "./dojo/tools/nikto/parser.py", line 85, in __init__
    self.process_endpoints(finding, ip)
  File "./dojo/tools/nikto/parser.py", line 102, in process_endpoints
    protocol = rhost.group(1)
AttributeError: 'NoneType' object has no attribute 'group'
Error in parser: 'NoneType' object has no attribute 'group'
Internal Server Error: /product/1/import_scan_results
Traceback (most recent call last):
  File "./dojo/engagement/views.py", line 624, in import_scan_results
    parser = import_parser_factory(file, t, active, verified)
  File "./dojo/tools/factory.py", line 117, in import_parser_factory
    parser = NiktoXMLParser(file, test)
  File "./dojo/tools/nikto/parser.py", line 85, in __init__
    self.process_endpoints(finding, ip)
  File "./dojo/tools/nikto/parser.py", line 102, in process_endpoints
    protocol = rhost.group(1)
AttributeError: 'NoneType' object has no attribute 'group'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "./dojo/product/views.py", line 648, in import_scan_results_prod
    return import_scan_results(request, pid=pid)
  File "./dojo/engagement/views.py", line 633, in import_scan_results
    return HttpResponseRedirect(reverse('import_scan_results', args=(eid,)))
  File "/usr/local/lib/python3.6/site-packages/django/urls/base.py", line 90, in reverse
    return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
  File "/usr/local/lib/python3.6/site-packages/django/urls/resolvers.py", line 673, in _reverse_with_prefix
    raise NoReverseMatch(msg)

**Additional context** (optional)

**Comment by valentijnscholten** [comment]
can you provide a sample file to reproduce this?

**Comment by AliElKhatteb** [comment]
[nikto.zip](https://github.com/DefectDojo/django-DefectDojo/files/5546007/nikto.zip)

**Comment by AliElKhatteb** [comment]
aslo samething happen with trufflehog 
[trufflehog.zip](https://github.com/DefectDojo/django-DefectDojo/files/5546012/trufflehog.zip) 
its working fine with owasp-dependency-check

**Comment by AliElKhatteb** [comment]
any updates

**Comment by madchap** [comment]
> aslo samething happen with trufflehog
> [trufflehog.zip](https://github.com/DefectDojo/django-DefectDojo/files/5546012/trufflehog.zip)
> its working fine with owasp-dependency-check

The trufflog file you provided does not seem to be a valid JSON format. The import fails because of that.

```
$ cat trufflehog|python -m json.tool
Extra data: line 18 column 1 - line 250 column 1 (char 3240 - 78772)
```

**Comment by damiencarol** [comment]
Took a look at the Nikto report. One of the URL is not-well-formed. I'm working on a PR to be able to load the report with not well formed URLs.
Working on a pull request to be able to import this not well formed reports: #4357

**Comment by valentijnscholten** [comment]
fixed in 1.15.0
